### DISCUSSION TITLE
Garbled Window Rendering when Connecting/Disconnecting to External Display on Mac

### DISCUSSION THREAD

**Discussion by JosephMart** [ghostty-org/ghostty#4034] [Feature Requests, Ideas] [answered]
![image](https://github.com/user-attachments/assets/254ce6a5-d148-4c1c-afde-c83bb5d31c0a)

## Summary
On a 14 in. Mac Sequoia 15.2, Ghostty windows fail to render terminal content properly when switching between displays if the app was in a non-visible state during the transition (e.g hidden behind other windows or in a different Desktop/Space). A keypress with Ghostty as the active window appears to force a repaint and resolves the issue.

## Repro Steps

Example with multiple Spaces.

1. On the MacBook's native display, create two Desktops/Spaces ([Learn more about multiple spaces](https://support.apple.com/guide/mac-help/work-in-multiple-spaces-mh14112/mac)).
2. Open Ghostty in Desktop A.
3. Switch to Desktop B.
4. Connect an external display to the MacBook.
5. Force Desktop B to appear on the external display (e.g., make the external display the main display or close the MacBook lid).

**Additional Note:** This issue also occurs in reverse (starting from the external display).

**Expected Behavior:** Terminal content renders correctly when the Ghostty window becomes visible on the new display.

**Observed Behavior:** Terminal content does not render until a keypress triggers a repaint.

## Device Info
MacBook Pro 14-inch, Nov 2024
Sequoia 15.2
MacBook Display Resolution - 1512x982

External Display Resolution - 3440x1440

**Comment by JakeWharton** [comment, marked as answer]
Tracked by https://github.com/ghostty-org/ghostty/issues/2731
